more_itertools/more.py

```diff
@@ -3335,23 +3333,23 @@ def set_partitions_helper(L, k):
         elif n == k:
             yield [[s] for s in L]
         else:
-            e, *M = L
-            for p in set_partitions_helper(M, k - 1):
+            e, *m = L
```

Comment:
Please stop making blind substitutions.  In classical and idiomatic choices of variable names, we use `m` and `n` for populations sizes, `k` and `r` for subpopulation sizes, `p` and `q` for paired variables, `i`, `j`, `k` for indices, etc.

In this case, using `m` makes the code harder to read.  If `M` has to go, then switch to longer names to accurately describe the code.  Perhaps `head` and `tail` for `e` and `M`.

Reply:
Thank you.